Assets/MixedRealityToolkit-SDK/Features/Audio/Influencers/AudioLoFiEffect.cs

```diff
@@ -0,0 +1,198 @@
+        /// </summary>
+        private AudioLoFiFilterSettings filterSettings;
+
+        /// <summary>
+        /// The filters used to simulate the source quality.
+        /// </summary>
+        private AudioLowPassFilter lowPassFilter;
+        private AudioHighPassFilter highPassFilter;
+
+        /// <summary>
+        /// Collection used to look up the filter settings that match the selected
+        /// source quality.
+        /// </summary>
+        private Dictionary<AudioLoFiSourceQualityType, AudioLoFiFilterSettings> sourceQualityFilterSettings =
+            new Dictionary<AudioLoFiSourceQualityType, AudioLoFiFilterSettings>();
+
+        private void Awake()
+        {
+            LoadQualityFilterSettings();
+
+            filterSettings = sourceQualityFilterSettings[SourceQuality];
+
+            lowPassFilter = gameObject.GetComponent<AudioLowPassFilter>();
+            lowPassFilter.cutoffFrequency = filterSettings.LowPassCutoff;
+
+            highPassFilter = gameObject.GetComponent<AudioHighPassFilter>();
+            highPassFilter.cutoffFrequency = filterSettings.HighPassCutoff;
+        }
+
+        private void Update()
+        {
+            AudioLoFiFilterSettings newSettings = sourceQualityFilterSettings[SourceQuality];
+            if (newSettings != filterSettings)
+            {
+                // If we have an attached AudioInfluencerController, we must let it know
+                // about our filter settings change, otherwise other effects may not behave
+                // as expected.
+                AudioInfluencerController influencerController = gameObject.GetComponent<AudioInfluencerController>();
+                if (influencerController != null)
+                {
+                    influencerController.SetNativeLowPassCutoffFrequency(newSettings.LowPassCutoff);
+                    influencerController.SetNativeHighPassCutoffFrequency(newSettings.HighPassCutoff);
+                }
+
+                filterSettings = newSettings;
+                lowPassFilter.cutoffFrequency = filterSettings.LowPassCutoff;
+                highPassFilter.cutoffFrequency = filterSettings.HighPassCutoff;
+            }
+        }
+
+        /// <summary>
+        /// Populates the source quality filter settings collection.
+        /// </summary>
+        private void LoadQualityFilterSettings()
+        {
+            if (sourceQualityFilterSettings.Keys.Count > 0) { return; }
+
+            sourceQualityFilterSettings.Add(
+                AudioLoFiSourceQualityType.FullRange,
+                new AudioLoFiFilterSettings(10, 22000));
+            sourceQualityFilterSettings.Add(
+                AudioLoFiSourceQualityType.NarrowBandTelephony,
+                new AudioLoFiFilterSettings(300, 3400));
+            sourceQualityFilterSettings.Add(
+                AudioLoFiSourceQualityType.WideBandTelephony,
+                new AudioLoFiFilterSettings(50, 7000));
```

Comment:
magic numbers?

Reply:
they are specific frequency ranges for the audio source. there is no real reason to make consts out of them for this useage.

Reply:
i added comments. i assumed (incorrectly?) that the quality type specified above clarified the question.

i dont think consts are that useful here since it is specific to the lo fi effect.